src/components/Stage.tsx

```diff
@@ -51,12 +60,17 @@ const Stage = () => {
         </mesh>
         {state.stage.eventType === "abilityRefresh" && (
           <AbilityTiles
-            isClickable={true}
-            canRefresh={true}
-            xStart={0}
+            isClickable={canRefresh}
+            canRefresh={canRefresh}
+            xStart={abilityTiles.length === 1 ? 0 : 0 - Math.ceil(abilityTiles.length / 2) * (abilityOffset / 2)}
             yStart={-cardHeight}
             zStart={50}
-            abilities={find(state.players, { uid: state.turn.player })!.abilities.filter((ability) => ability.isUsed)}
+            abilities={
+              canRefresh
+                ? filter(player.abilities, { isUsed: true })
+                : filter(player.abilities, { uid: last(state.turn.refreshedAbilityUids) })
+            }
+            orientation="horizontal"
```

Comment:
Feels like `<AbilityTiles>` idea doesn't really align with other stuff we have. I think I'll refactor it to a flatter architecture in **separate PR**

```jsx
{abilities.map((ability)=>
      <AbilityTile
            isClickable={emit.tokenClick(ability)}
            xStart={abilityTiles.length === 1 ? 0 : 0 - Math.ceil(abilityTiles.length / 2) * (abilityOffset / 2)}
            yStart={-cardHeight}
            zStart={50}
            ability={ability}>
)}
```

Reply:
Yeah agreed👍🏻